cloud/aws/beanstalk/monitors-beanstalk.tf

```diff
@@ -0,0 +1,136 @@
+	EOF
+
+	rule {
+		description           = "is too high >= ${var.health_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.health_disabled_critical, var.health_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.health_notifications_critical, var.health_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "is too high >= ${var.health_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.health_disabled_warning, var.health_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.health_notifications_warning, var.health_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "latency_p90" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] AWS beanstalk applicaion latency p90"
+
+	program_text = <<-EOF
+		signal = data('ApplicationLatencyP90', filter=filter('namespace', 'AWS/ElasticBeanstalk') and filter('stat', 'lower') and filter('InstanceId', '*') and ${module.filter-tags.filter_custom})${var.latency_p90_aggregation_function}.${var.latency_p90_transformation_function}(over='${var.latency_p90_transformation_window}')
+		detect(when(signal >= ${var.latency_p90_threshold_critical})).publish('CRIT')
+		detect(when(signal >= ${var.latency_p90_threshold_warning})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "is too high > ${var.latency_p90_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.latency_p90_disabled_critical, var.latency_p90_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.latency_p90_notifications_critical, var.latency_p90_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "is too high > ${var.latency_p90_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.latency_p90_disabled_warning, var.latency_p90_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.latency_p90_notifications_warning, var.latency_p90_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "5xx_error_rate" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] AWS beanstalk 5xx error rate"
```

Comment:
```suggestion
 	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] AWS Beanstalk 5xx errors rate"
```

Reply:
Will correct Beanstalk so the B is capital. but the correct english is error rate not errors rate